ecli/client/src/http_client.rs

```diff
@@ -0,0 +1,187 @@
+pub struct StopCommand {
+    #[clap(required = true, help = "ID of the task")]
+    pub id: ProgramHandle,
+}
+
+#[derive(Parser)]
+pub struct PauseCommand {
+    #[clap(required = true, help = "ID of the task")]
+    pub id: ProgramHandle,
+}
+
+#[derive(Parser)]
+pub struct ResumeCommand {
+    #[clap(required = true, help = "ID of the Task")]
+    pub id: ProgramHandle,
+}
+
+pub(crate) async fn handle_client_command(cmd: ClientCmd) -> EcliResult<()> {
+    let client = EcliHttpClient::new(cmd.opts.endpoint)?;
+    match cmd.cmd {
+        ClientSubCommand::Start(StartCommand {
+            prog,
+            extra_args,
+            name,
+            export_json,
+            prog_type: user_prog_type,
+        }) => {
+            let (buf, prog_type) = load_prog_buf_and_guess_type(&prog, user_prog_type).await?;
+
+            let handle = client
+                .start_program(name, &buf, prog_type, export_json, &extra_args, None)
+                .await?;
+            println!("{}", handle);
+        }
+        ClientSubCommand::Stop(StopCommand { id }) => {
+            client.terminate_program(id).await?;
+        }
+
+        ClientSubCommand::Pause(PauseCommand { id }) => {
+            client.set_program_pause_state(id, true).await?;
+        }
+        ClientSubCommand::Resume(ResumeCommand { id }) => {
+            client.set_program_pause_state(id, false).await?;
+        }
+        ClientSubCommand::List => {
+            for item in client.get_program_list().await? {
+                println!("{} {} {:?}", item.id, item.name, item.status);
+            }
+        }
+        ClientSubCommand::Log(LogCommand { id, follow, cursor }) => {
+            if follow {
+                let mut last_poll = None;
+                loop {
+                    let logs = client.fetch_logs(id, last_poll, None).await?;
+                    for (cursor, log) in logs.into_iter() {
+                        println!("{}", log.log);
+                        last_poll = Some(cursor + 1);
+                    }
+                    tokio::time::sleep(Duration::from_millis(100)).await;
+                }
+            } else {
+                let logs = client.fetch_logs(id, cursor, None).await?;
+                let mut last_cursor = None;
+                for (cursor, log) in logs.into_iter() {
+                    last_cursor = Some(cursor);
+                    match log.log_type {
+                        ecli_lib::runner::LogType::Stdout => {
+                            print!("{}", log.log);
+                        }
+                        ecli_lib::runner::LogType::Stderr => {
+                            eprint!("{}", log.log);
+                        }
+                        ecli_lib::runner::LogType::Plain => {
+                            println!("<{}> {}", log.timestamp, log.log);
+                        }
+                    }
+                }
+                if let Some(v) = last_cursor {
+                    println!();
```

Comment:
- The new line is needed, since the logs previously printed may not be ended with a new line
- Originally I think format strings like "\nNo logs fetched" looks ugly and confusing. what do you think?

Reply:
I think it should be a common practice, see
https://github.com/search?q=println%21%28%22%5Cn&type=code

Reply:
> I think it should be a common practice, see https://github.com/search?q=println%21%28%22%5Cn&type=code

Ok. I'll apply it

Reply:
Fixed now

Reply:
But there is another problem: Different platforms have dirrerent line breaks chars, `\r\n` on Windows and `\n` on *nix. Remote clients can be run under Windows. Should we take this plarform-specified behavior into consideration? on just use `\n` directly?

Reply:
> But there is another problem: Different platforms have dirrerent line breaks chars, `\r\n` on Windows and `\n` on *nix. Remote clients can be run under Windows. Should we take this plarform-specified behavior into consideration? on just use `\n` directly?

Infact, use `println!()` is a good way to sole such platform differences

Reply:
sorry, I don't know this diff on Windows. In that case, let's keep the original way.

Reply:
Ok, I added that println!() back